src/pages/home/report/ReportActionCompose.js

```diff
@@ -207,9 +207,11 @@ class ReportActionCompose extends React.Component {
                         )}
                     </AttachmentModal>
                     <TouchableOpacity
-                        style={[styles.chatItemSubmitButton, styles.buttonSuccess]}
+                        style={[styles.chatItemSubmitButton,
+                            this.props.comment ? styles.buttonSuccess : styles.buttonDisable]}
                         onPress={this.submitForm}
                         underlayColor={themeColors.componentBG}
+                        disabled={this.props.comment.length === 0}
```

Comment:
Using `this.props.comment` does create the intended behavior for both situations. But I think you were on a better path before as `this.props.comment` is updating after a delay. The initial color change of the button was **faster** before. By returning to the original solution and setting state in `componentDidUpdate()` you can solve this problem.

Reply:
@marcaaron Thanks for this, and i have changed the code with  `this.State` Approach, to fix the delay issue. Please find it in the latest commit, And do let me know if there is anything else you want me to do, waiting for your feedbacks.